### PR TITLE
User interface

### PR DESCRIPTION
**Pull request by misterpaul4**
Made only the interface for tic tac toe without game logic , game logic was made on milestone 3

### CODE REVIEW

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @misterpaul4 and @Ferm0494 
### You have both done a good job so far :clap: However, you have a few issues to you need to fix.
- [ ] You need a descriptive pull request detailing what you have done on this milestone
- [ ] The does not determine whether its a win, a loss or a draw, it just terminates

### Make these changes and submit another code review. Good luck.
- Happy coding!

By Raymond
slack @iamraymond

**Comment by Ferm0494** [comment]
Hello @rOluochKe the milestone 2 says exactly this can you please help us understand : 

_Display prompts for users (hint: puts/print), collect user input (hint: get.chomp) and only mock the basic flow of the game (you can use puts or inline comments to say for example “Now, your move is displayed on the board”).
Make sure that you can run your game with your executable file and play with the basic flow.
The final point of this milestone is to make sure that you thought about all actions needed to play Tic Tac Toe. **It does not need to work yet** - this will be done in the next milestone._

✅ A descriptive pull request was given.

File: bin/main [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    puts 'Wrong input'.red
+  end
+end
+
+def start_game()
+  puts "Positions size : #{$positions.size}"
+  while $exit.eql? 0 and $winner.eql? 0 and !$count.eql? $positions.size
+    board
+    user_input
+  end
```
Comment (mcrd25) [inline bin/main:59]: After this line is a good place to add your comment etc.

**Review by mcrd25 — CHANGES_REQUESTED** [review]
# Changes Required 🔍 

Hi Team!

Yes you are absolutely right! Your application doesn't to have to work at this point!
However, you do have to show that you are handling the output of the game status when it's over. As in either a comment or puts statement that states, "Game over: draw" etc. So simply adding a commented saying something like `# displays game status winner or draw` or something similar to show that you thought about it. 


I hope this review helps! Thanks for the improved PR description. 🙂 

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Comment by Anna-Myzukina** [comment]
# Hi team! :clap: 
####  You are missing one requirement  in Milestone 1 : 

*  In your project should be `bin` directory with `main.rb` file  from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)
screenshot
*  `main.rb` is [executable file](https://commandercoriander.net/blog/2013/02/16/making-a-ruby-script-executable/):

* To Test if this executable file works with next command: 
`bin/main.rb` or `ruby bin/main.rb` and it should displays anything.

### Good job in Milestone 2 :eyes: 

### *The final point of this milestone is to make sure that you thought about all the actions needed to play Tic Tac Toe from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)

### Basic game flow is prepared : 

- [ ] [REQUIRED] game informs player if the selected move is a winning move

- [ ] [REQUIRED] game informs player if the selected move is a draw move
 

### [SUGGESTION] The structure of the game's loop should be as follows:

##### Implementation details

* [ ] A loop is needed in this game (recommended is `while` but anything else will do) ([OdinProject hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/loop.png))
* [ ] Inside loop, you can all actions repeated for each move
* [ ] A loop needs to be terminated at some point (after a winning move or draw to be more specific). It can be done with variable:

Using out-of-nowhere numbers to terminate a loop, for example 5.times do ... end is a signal that something is wrong. We recommend the usage of a terminating variable (while game_on) 

### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*
*Happy coding * <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Comment by Ferm0494** [comment]
Hi, @Anna-Myzukina  Thanks for the review 👍 , I will list the requirements directly from the pathwright: 

1. Display prompts for users (hint: puts/print), collect user input (hint: get.chomp) and only mock the basic flow of the game (you can use puts or inline comments to say for example “Now, your move is displayed on the board”).

2.  Make sure that you can run your game with your executable file and play with the basic flow.
The final point of this milestone is to make sure that you thought about all actions needed to play Tic Tac Toe. It does not need to work yet - this will be done in the next milestone.

- There's already an executable file on the **bin**, please check that since the command works, and also the milestone1 has been **approved** 

- Please read @mcrd25 comments as well, since the **game logic is only for milestone 3**, the game now terminates with a **Game Over** since **specific winner and draw is part of the next milestone**.

**Comment by lekegitrepo** [comment]
#### Hi @misterpaul4 and @Ferm0494 :clap: :clap:
#### Well done, you have done a good job. However, as mentioned in the last review; these two implements are part of the checklist that must be considered before this project (milestone 2) can be approved :point_down:

* [ ] `[REQUIRED]` game informs player if the selected move is a winning move
* [ ] `[REQUIRED]` game informs player if the selected move is a draw move

**_kindly implement the required changes and submit for review._**

### _Reviewed by Temitayo Adeleke._

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
#### Status: Request changes

**Comment by Ferm0494** [comment]
@lekegitrepo That's for **milestone3** please read the requirements carefully from this project in here Thanks.

**Milestone 2: User interface**
Suggested branch name: user_interface.
main.rb file should be responsible for user interaction. In this milestone you should only create content in this file.
Think about steps needed to play the game:
what do you need from player?
what do you need to display for player?
Display prompts for users (hint: puts/print), collect user input (hint: get.chomp) and only mock the basic flow of the game (you can use puts or inline comments to say for example “Now, your move is displayed on the board”).
Make sure that you can run your game with your executable file and play with the basic flow.
The final point of this milestone is to make sure that you thought about all actions needed to play Tic Tac Toe. It does not need to work yet - this will be done in the next milestone.

**Comment by lekegitrepo** [comment]
#### @misterpaul4 and @Ferm0494 :100: :1st_place_medal:
#### Good job. This project is approved :climbing:

### Reviewed by Temitayo Adeleke.

**Review by lekegitrepo — APPROVED** [review]
#### Status: Approved.

**Pull request merged**